tests/ResponseDocumentingTest.php

```diff
@@ -96,3 +98,57 @@ public function index()
         return response()->json(['foo' => 'bar']);
     }
 }
+
+test('laravel responder success support', function () {
+    \Illuminate\Support\Facades\Route::get('api/test', [Foo_TestFive::class, 'index']);
+
+    \Dedoc\Scramble\Scramble::routes(fn (Route $r) => $r->uri === 'api/test');
+    $openApiDocument = app()->make(\Dedoc\Scramble\Generator::class)();
+
+    assertMatchesSnapshot($openApiDocument);
+});
+class Foo_TestFive
+{
+    public function index()
+    {
+        $user = [];
+        return responder()->success($user, UserTransformer::class)->respond(201, [
+            'Authorization' => 'Bearer ghjkiuygvbnmkuyg'
+        ]);
+    }
+}
+
+/**
+ * @mixin User
+ */
+class UserTransformer extends TransformerAbstract
+{
+    public function transform(User $user): array
+    {
+        return [
+            'id' => $user->id,
+            'name' => $user->name,
+            'testString' => 'iasdihas',
+            'testNumber' => 2333
+        ];
+    }
+}
+
+test('laravel responder error support', function () {
+    \Illuminate\Support\Facades\Route::get('api/test', [Foo_TestFive::class, 'index']);
+
+    \Dedoc\Scramble\Scramble::routes(fn (Route $r) => $r->uri === 'api/test');
+    $openApiDocument = app()->make(\Dedoc\Scramble\Generator::class)();
+
+    assertMatchesSnapshot($openApiDocument);
+});
+class Foo_TestSix
+{
+    public function index()
```

Comment:
```suggestion
    public function index(): UserTransformer
```

Reply:
A controller can't return a UserTransformer, it returns a json Resource, however I did not specify it because the point is for the library to infer it automatically

Reply:
Gotcha!